pkg/unifiedpush/client.go

```diff
@@ -54,30 +54,30 @@ type UnifiedpushClient struct {
 }
 
 // GetApplication does a GET for a given PushApplication based on the PushApplicationId
-func (c UnifiedpushClient) GetApplication(p *pushv1alpha1.PushApplication) (string, error) {
-	if p.Status.PushApplicationId == "" {
+func (c UnifiedpushClient) GetApplication(p *pushv1alpha1.PushApplication) (PushApplication, error) {
+	if p.ObjectMeta.Annotations["pushApplicationId"] == "" {
 		// We haven't created it yet
-		return "", nil
+		return PushApplication{}, nil
 	}
```

Comment:
does this actually backward compatible? Should we also check if the id is in the status field too? Just thinking about existing apps, they will only have id in the status field?

Another option would be to add the annotation if the application id only exists in the status field but not in annotations?

Reply:
I think the updated code here does what you are talking about.